src/test/kotlin/edu/carroll/polyominoes/service/account/RegisterServiceImplTest.kt

```diff
@@ -0,0 +1,289 @@
+    }
+
+    @Test
+    fun createUserBlankPasswordTest() {
+        assertFalse(registerServiceImpl.createUser(testAccount1.username, testAccount1.email,"    "))
+    }
+    @Test
+    fun createUserTabPasswordTest() {
+        assertFalse(registerServiceImpl.createUser(testAccount1.username, testAccount1.email,"\t"))
+    }
+    @Test
+    fun createUserNewlinePasswordTest() {
+        assertFalse(registerServiceImpl.createUser(testAccount1.username, testAccount1.email,"\n"))
+    }
+
+    @Test
+    fun validateUsernameExistOneSuccessTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+    }
+
+    @Test
+    fun validateUsernameExistTwoSuccessTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount2.username))
+    }
+
+    @Test
+    fun validateUsernameExistDuplicateSuccessTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+    }
+
+    @Test
+    fun validateUsernameExistTwoDuplicateSuccessTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount2.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount2.username))
+    }
+
+    @Test
+    fun validateUsernameExistFailsAfterUsernameExistsTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertTrue(registerServiceImpl.createUser(testAccount1.username, testAccount1.email, testAccount1.rawPassword))
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount1.username))
+    }
+
+    @Test
+    fun validateUsernameExistFailsAfterUsernamesExistsTest() {
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertFalse(registerServiceImpl.validateUsernameExist(testAccount2.username))
+        assertTrue(registerServiceImpl.createUser(testAccount1.username, testAccount1.email, testAccount1.rawPassword))
+        assertTrue(registerServiceImpl.createUser(testAccount2.username, testAccount2.email, testAccount2.rawPassword))
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount2.username))
+    }
+
+    @Test
+    fun validateUsernameExistInvalidTest() {
+        assertTrue(registerServiceImpl.createUser(testAccount1.username, testAccount1.email, testAccount1.rawPassword))
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount1.username))
+    }
+
+    @Test
+    fun validateUsernameExistTwoUsernamesInvalidTest() {
```

Comment:
Why does have the name 'UsernamesInvalid' in the method name, since both usernames appear to be valid?